### PR TITLE
Migrate blob tx

### PR DESCRIPTION
**Pull request by WorldDogs**

### CODE REVIEW

File: tx-submitter/services/rollup.go [hunk 1]
```diff
@@ -423,47 +426,111 @@ func (sr *SR) rollup() error {
...
+		versionedHashes := make([]common.Hash, 0)
+		for _, commit := range batch.Sidecar.Commitments {
+			versionedHashes = append(versionedHashes, kZGToVersionedHash(commit))
+		}
+		tx = types.NewTx(&types.BlobTx{
+			ChainID:    uint256.MustFromBig(sr.chainId),
+			Nonce:      nonce,
+			GasTipCap:  uint256.MustFromBig(big.NewInt(tip.Int64())),
+			GasFeeCap:  uint256.MustFromBig(big.NewInt(gasFeeCap.Int64())),
+			Gas:        5000000,
```
Comment (FletcherMan) [inline tx-submitter/services/rollup.go:459]: need estimate gas limit for blob tx

File: tx-submitter/services/rollup.go [hunk 2]
```diff
@@ -239,26 +241,21 @@ func (sr *SR) finalize() error {
...
-				"finalize_cnt", finalizeCnt,
-			)
+		} else if utils.ErrStringMatch(err, core.ErrGasLimit) { // ErrGasLimit
+			log.Error("tx exceeds block gas limit", "gas", newSignedTx.Gas(), "finalize_cnt", finalizeCnt)
 			return fmt.Errorf("send tx error:%v", err.Error())
-		case core.ErrNonceTooLow.Error():
+		} else if utils.ErrStringMatch(err, core.ErrNonceTooLow) { //ErrNonceTooLow
 			return fmt.Errorf("send tx error:%v", err.Error())
-		default:
+		} else {
```
Comment (curryxbo) [inline tx-submitter/services/rollup.go:258]: "Else if" and "Else" don't seem so different

File: tx-submitter/services/rollup.go [hunk 3]
```diff
@@ -414,7 +413,11 @@ func (sr *SR) rollup() error {
 		Signature:              *signature,
 	}
 
-	var signedTx *types.Transaction
+	calldata, err := sr.abi.Pack("commitBatch", rollupBatch, uint32(minGasLimit))
+	if err != nil {
+		return fmt.Errorf("pack calldata error:%v", err)
+	}
+	// var signedTx *types.Transaction
```
Comment (curryxbo) [inline tx-submitter/services/rollup.go:420]: pls delete useless code

File: tx-submitter/services/rollup.go [hunk 4]
```diff
@@ -423,47 +426,111 @@ func (sr *SR) rollup() error {
...
 			log.Error("tx exceeds block gas limit", "gas", newSignedTx.Gas(), "chunks_len", len(batch.Chunks))
 			return fmt.Errorf("send tx error:%v", err.Error())
-		case core.ErrNonceTooLow.Error():
+		} else if utils.ErrStringMatch(err, core.ErrNonceTooLow) { //ErrNonceTooLow
 			return fmt.Errorf("send tx error:%v", err.Error())
-		default:
+		} else {
 			return fmt.Errorf("send tx error:%v", err.Error())
 		}
+
```
Comment (curryxbo) [inline tx-submitter/services/rollup.go:514]: ditto

File: tx-submitter/services/rollup.go [hunk 5]
```diff
@@ -423,47 +426,111 @@ func (sr *SR) rollup() error {
...
+		// 	receipt, newSignedTx, err = sr.replaceTx(newSignedTx)
+		// 	if err != nil {
+		// 		return fmt.Errorf("replace tx error:%v", err)
+		// 	} else {
+		// 		log.Info("replace tx success")
+		// 	}
+		// case core.ErrGasLimit.Error():
+		// 	log.Error("tx exceeds block gas limit", "gas", newSignedTx.Gas(), "chunks_len", len(batch.Chunks))
+		// 	return fmt.Errorf("send tx error:%v", err.Error())
+		// case core.ErrNonceTooLow.Error():
```
Comment (curryxbo) [inline tx-submitter/services/rollup.go:529]: ditto

**Review by FletcherMan — APPROVED** [review]

**Pull request merged**